library/platform_util.c

```diff
@@ -172,50 +183,77 @@ int mbedtls_platform_memcmp( const void *buf1, const void *buf2, size_t num )
     return( (int) diff | (int) ( flow_counter ^ num ) );
 }
 
-uint32_t mbedtls_platform_random_uint32( )
+/* This function implements a non-cryptographic random number generator based
+ * on the linear congruential generator algorithm. Additionally, if the
+ * MBEDTLS_ENTROPY_HARDWARE_ALT flag is defined, the seed is changed with using
+ * a hardware random number generator every RAND_SEED_LIFE number of iterations.
+ * 
+ * The value of the returned number is in the range [0; 0x80000000).
+ */
+static uint32_t mbedtls_platform_random( void )
```

Comment:
Why is such range used? It causes a need to call `mbedtls_platform_random_uint32` twice and merge the results.

Reply:
This is due to the use of the modulo 0x80000000 value. This is a standard value used by glibc and ANSI C